### ISSUE TITLE
CA1416 (Validate platform compatibility) doesn't work for implicit operators

### ISSUE THREAD

**Issue by LunarWhisper** [dotnet/roslyn-analyzers#7350]
### Analyzer

**Diagnostic ID**: [CA1416](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1416): `Validate platform compatibility`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://learn.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 8.0.301](https://dotnet.microsoft.com/download/dotnet/8.0)

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: 9.0.0-preview.24318.1

### Describe the bug

Any operators can be marked with the `SupportedOSPlatform` attribute, but in fact it is not validated in any way and does not affect anything.

### Steps To Reproduce

```csharp
internal class Program
{
    [SupportedOSPlatform("linux")]
    public static void Main(string[] args)
    {
        Program instance = new();
        
        int value0 = (int)instance;
        string value1 = instance;
        string value2 = AsString();
        bool flag1 = instance == instance;
        bool flag2 = instance == instance;
    }

    [SupportedOSPlatform("windows")]
    public static explicit operator int(Program program) => 1;

    [SupportedOSPlatform("windows")]
    public static implicit operator string(Program program) => AsString();
    
    [SupportedOSPlatform("windows")]
    public static bool operator ==(Program x, Program y) => false;
    
    [SupportedOSPlatform("windows")]
    public static Boolean operator !=(Program x, Program y) => !(x == y);

    [SupportedOSPlatform("windows")]
    private static string AsString() => "Hello World";
}
```

### Expected behavior

These lines should be highlighted as warning, because this operator can only be used in Windows.
Only the conversion operators have meaning, the rest are added for the company.
```csharp
        int value0 = (int)instance;
        string value1 = instance;
        bool flag1 = instance == instance;
        bool flag2 = instance == instance;
```

### Actual behavior

Only `string value2 = AsString();` highlighted as warning.

## Additional context

Use case: Allow implicit casting of LocalAbsolutePath to WindowsAbsolutePath on Windows only.
